components/studio/libexec/hab-studio-type-default.sh

```diff
@@ -66,10 +66,28 @@ finish_setup() {
     return 0
   fi
 
+  # In the overwhelming majority of instances, you will want to use
+  # stable components in your Studio.
+  #
+  # However, the Habitat team needs to build studios using unstable
+  # components in order to fully test things. This `CI_OVERRIDE_CHANNEL`
+  # is supplied as a back-door to enable that to happen. It should be
+  # set to the name of a channel to preferentially pull packages from.
+  #
+  # If you're not on the Habitat core team, you will likely never need
+  # this, or even have to know it exists.
+  #
+  # (This is also why we're not using HAB_BLDR_CHANNEL for this and
+  # replicating the fallback logic from hab-plan-build; it'd be too
+  # easy to create an unstable studio.)
   for pkg in $pkgs; do
-    _hab install $pkg
+    if _hab install --channel="${CI_OVERRIDE_CHANNEL}" ${pkg}; then
```

Comment:
```
if [[ -n "${CI_OVERRIDE_CHANNEL}" ]]; then
   _hab install --channel="${CI_OVERRIDE_CHANNEL}" ${pkg}
...
```

Would something like ^ work as well (instead of executing `_hab install ...` twice)? That would also avoid doubled log message:

```
» Installing core/hab-backline/0.51.0 from channel ''
» Installing core/hab-backline/0.51.0 from channel 'stable'
```

Reply:
Given that this may be sourced with `nounset` (aka `set -u`), changing to this should suffice:
```bash
if _hab install --channel="${CI_OVERRIDE_CHANNEL:-}" ${pkg}; then
    info "Override channel set; installed ${pkg} from ${CI_OVERRIDE_CHANNEL}"
else
    _hab install ${pkg}
fi
```
It will still result in the doubled log message (which may be useful diagnostics or noise depending on how you look at it). Compared to
```bash
if [[ -n "${CI_OVERRIDE_CHANNEL:-}" ]]; then
   _hab install --channel="${CI_OVERRIDE_CHANNEL}" ${pkg}
else
    _hab install ${pkg}
fi
```
there is a meaningful semantic difference. The first option will fall back to the default channel for `_hab install` whereas the second will cause the entire script to fail if the install with 
`CI_OVERRIDE_CHANNEL` fails. I'm not sure what the intention is.